edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs

```diff
@@ -75,11 +75,42 @@ impl Client {
 				serde_json::from_slice(&*response)
 				.map_err(|err| Error::GetServerRootCertificate(ApiErrorReason::ParseResponseBody(Box::new(err))))?;
 
-			let certificate =
-				native_tls::Certificate::from_pem(certificate.as_bytes())
-				.map_err(|err| Error::GetServerRootCertificate(ApiErrorReason::ParseResponseBody(Box::new(err))))?;
+			let mut server_root_certificate = vec![];
+
+			let mut current_cert = String::new();
+			let mut lines = certificate.lines();
+	
+			if lines.next() != Some("-----BEGIN CERTIFICATE-----") {
```

Comment:
Can you refactor this into a method that you can then unit test?

Reply:
that's a good idea. Making a note of it